config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -200,6 +204,8 @@ private static Object getListValue(Class<?> beanClass, Type parameterType, Class
                 beanList.add(createInternal(listMember, (Class<?>) elementType));
             }
             return beanList;
+        } else if (((Class<?>) elementType).isEnum()) {
+            return config.getEnumList((Class<Enum>) elementType, configPropName);
```

Comment:
Yes, in theory, an enum class may have a bean property. So, it makes sense to move the enum case before the "at least one property" case.